File: UI/financial_manager/src/app/models/transaction.ts
import { Category } from './category';
import { User } from './user';

export class Transaction {
  id?: number;
  title: string;
  significance: number;
  transactionType: string;
  createdAt?: Date;
  expenseDate?: Date;
  category?: Category;
  user?: User;

  constructor(
    title: string,
    significance: number,
    transactionType: string,
    expenseDate?: Date,
    category?: Category,
    user?: User
  ) {
    this.title = title;
    this.significance = significance;
    this.transactionType = transactionType;
    this.expenseDate = expenseDate;
    this.category = category;
    this.user = user;
  }
}

File: financial-manager-api/Entities/Models/Transaction.cs
using System;
using System.Collections.Generic;
using financial_manager.Entities.Shared;

namespace financial_manager.Entities.Models;

public partial class Transaction : BaseModel
{
    public int UserId { get; set; }

    public string Title { get; set; } = null!;

    public int? CategoryId { get; set; }

    public decimal Significance { get; set; }

    public string TransactionType { get; set; } = null!;

    public DateTime? ExpenseDate { get; set; }

    public DateTime CreatedAt { get; set; }

    public virtual Category? Category { get; set; }

    public virtual User User { get; set; } = null!;
}
